### PR TITLE
Fix Gradle 7 compatibility

### PR DESCRIPTION
**Pull request by fawind**
## Before this PR
The plugin was not compatible with Gradle 7 as the underlying jengelman shadowjar plugin was not compatible.

## After this PR
Update underlying shadowjar plugin. Note that this increases the minimum required Gradle version to 7.0.

==COMMIT_MSG==
Fix Gradle 7 compatibility
==COMMIT_MSG==

## Possible downsides?
New minimum Gradle version is 7. So we'll have a hard split between 6.x users having to use the old version and 7.x the new version.

### CODE REVIEW

**Comment by fawind** [comment]
I guess we need to exclude this version from the regular build dependency upgrade excavator until we have a decent Gradle 7 adoption as other updates will get stuck otherwise.

File: changelog/@unreleased/pr-137.v2.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+type: break
+break:
+  description: Fix Gradle 7 compatibility. Increase minimum required Gradle version
+    to 7.0.
```
Comment (CRogers) [inline changelog/@unreleased/pr-137.v2.yml:4]: It's super unfortunate we need to split the versions like this. I think we might technically be able to support both by doing something like shading two different versions of the shadow plugin, or by having two different artifacts for this plugin (one <7 one >=7) but both are pretty grim.

I think with the excavator we can just choose to either keep the version pinned if `GradleVersionNumber.current()` is <7 or use latest if >7. People will just have to upgrade to 7 if they want a bug fix.

File: src/test/groovy/com/palantir/gradle/shadowjar/ShadowJarPluginIntegrationSpec.groovy [hunk 2]
```diff
@@ -400,10 +399,11 @@ class ShadowJarPluginIntegrationSpec extends IntegrationSpec {
     }
 
     private String dependenciesInPom() {
-        def pomXml = new XmlParser().parse(
-                new File(projectDir, "${MAVEN_ROOT}/com/palantir/bar-baz_quux/asd-fgh/2/asd-fgh-2.pom"))
+        def pomFile = new File(projectDir, "${MAVEN_ROOT}/com/palantir/bar-baz_quux/asd-fgh/2/asd-fgh-2.pom")
+        def pomXml = new groovy.xml.XmlParser().parse(pomFile)
         // GCV publishes the shaded constraints in dependencyManagement - this should be fine
-        def dependenciesText = pomXml.dependencies.collect {XmlUtil.serialize(it)}.join('\n')
+        // Explicitly cast to "Node" before serializing to avoid errors due to Groovy choosing the wrong method override
```
Comment (CRogers) [inline src/test/groovy/com/palantir/gradle/shadowjar/ShadowJarPluginIntegrationSpec.groovy:405]: I was wondering when we'd hit the groovy 3.0 issues :p That reminds me, with Gradle 7/Groovy 3 we can start using `.stripIndent(true)` in our tests and have them work on a 15 JVM without weird formatting issues.

**Review by CRogers — APPROVED** [review]
G2G - Juts put a do not merge onto it in case you wanted to get the excavator piece ready pre merge.

**Comment by fawind** [comment]
Updated the excavator to pin to 1.4.0 for < Gradle 7

**Pull request merged**